### ISSUE TITLE
downloadPage doesn't errback when connection gets lost while downloading a file.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3645]
|<img alt="whirm's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| whirm reported|
|-|-|
|Trac ID|trac#3645|
|Type|defect|
|Created|2009-02-02 18:12:20Z|
|Branch|https://github.com/twisted/twisted/tree/incomplete-downloadpage-3645|

Deferred returned by downloadPage will not call its errbacks if connection is lost (server side or stopping client reactor), instead its callbacks are fired.

<details><summary>Searchable metadata</summary>

```
trac-id__3645 3645
type__defect defect
reporter__whirm whirm
priority__normal normal
milestone__ 
branch__branches_incomplete_downloadpage_3645 branches/incomplete-downloadpage-3645
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__web web
keywords__ 
time__1233598340000000 1233598340000000
changetime__1235952252000000 1235952252000000
version__None None
owner__ 
cc__truewhirm@...
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [26230]) Branching to 'incomplete-downloadpage-3645'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @exarkun|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)<a name="note_5"></a>|@jerub set owner to @exarkun|
|-|-|

Simple enough changes, and the unit tests are good. Thanks! Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set status to `closed`|
|-|-|

(In [26326]) Merge incomplete-downloadpage-3645

Author: exarkun
Reviewer: jerub
Fixes: [#3645](https://github.com/twisted/twisted/issues/3645)

Handle premature lost connections in `downloadPage` by errbacking the returned
`Deferred` instead of calling it back to indicate success.  Also handle errors
when closing the file in the error case.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
